src/ui/public/visualize/loader.js

```diff
@@ -1,75 +1,170 @@
-import $ from 'jquery';
-import uiRoutes from 'ui/routes';
+/**
+ * IMPORTANT: If you make changes to this API, please make sure to check that
+ * the docs (docs/development/visualize/development-create-visualization.asciidoc)
+ * are up to date.
+ */
+import angular from 'angular';
+import chrome from 'ui/chrome';
 import 'ui/visualize';
+import visTemplate from './loader_template.html';
+
+
+/**
+ * A handler to the embedded visualization. It offers several methods to interact
+ * with the visualization.
+ */
+class EmbeddedVisualizeHandler {
+  constructor(element, scope) {
+    this._element = element;
+    this._scope = scope;
+    this._renderComplete = new Promise(resolve => {
+      this._element.on('renderComplete', resolve);
+    });
+  }
+
+  /**
+   * Destroy the underlying Angular scope of the visualization. This should be
+   * called whenever you remove the visualization.
+   */
+  destroy() {
+    this._scope.$destroy();
+  }
+
+  /**
+   * Return the actual DOM element (wrapped in jQuery) of the rendered visualization.
+   * This is especially useful if you used `append: true` in the parameters where
+   * the visualization will be appended to the specified container.
+   */
+  get element() {
```

Comment:
I'm not a fan of native getters in Javascript, at least not now. They really hurt navigability in most IDEs, and do not make transparent if you are getting a direct reference to an object, or copies.

//e.g. property access in Javascript is pretty straightforward. But with getters we introduce pseudo-properties, which are just unnecessarily convoluted (especially because IDEs aren't good at picking up on it
```
if (myInstance.myProperty === myInstance.myProperty){
   //?!?!?! maybe, maybe not.
}
```
When using a method call , e.g. `myInstance.getMyProperty()`, users don't make as many assumptions. The advantage is that most Javascript IDEs are getting pretty decent at navigating across types when using methods as opposed to property-access.

Reply:
My main reason to use a property was, to give read only access to the element. So I am fine by replacing it with a `getElement()` method.